eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/ManagementTypeFactory.cs

```diff
@@ -67,24 +67,43 @@ private string BuildResourceProviderName()
         /// <inheritdoc/>
         protected override CSharpType? CreateCSharpTypeCore(InputType inputType)
         {
-            if (inputType is InputModelType model && KnownManagementTypes.TryGetManagementType(model.CrossLanguageDefinitionId, out var replacedType))
+            if (inputType is InputModelType model)
             {
-                return replacedType;
+                CSharpType? replacedType;
+
+                if (KnownManagementTypes.TryGetManagementType(model.CrossLanguageDefinitionId, out replacedType))
+                {
+                    return replacedType;
+                }
+
+                if (InputModelReplacement.Instance.TryFindModelReplacementType(model.CrossLanguageDefinitionId, out replacedType))
+                {
+                    return replacedType;
+                }
             }
             else if (inputType is InputPrimitiveType primitiveType && KnownManagementTypes.TryGetPrimitiveType(primitiveType.CrossLanguageDefinitionId, out var csharpType))
             {
                 return csharpType;
             }
+
             return base.CreateCSharpTypeCore(inputType);
         }
 
         /// <inheritdoc/>
         protected override ModelProvider? CreateModelCore(InputModelType model)
         {
-            if (KnownManagementTypes.TryGetManagementType(model.CrossLanguageDefinitionId, out var replacedType))
+            CSharpType? replacedType;
+
+            if (KnownManagementTypes.TryGetManagementType(model.CrossLanguageDefinitionId, out replacedType))
             {
                 return new InheritableSystemObjectModelProvider(replacedType.FrameworkType, model);
             }
+
+            if (InputModelReplacement.Instance.TryFindModelReplacementType(model.CrossLanguageDefinitionId, out replacedType))
+            {
+                return new InheritableSystemObjectModelProvider(replacedType.FrameworkType, model);
+            }
```

Comment:
so in our initial design, we call this thing **Inheritable**SystemObjectModelProvider and make it to inherit from `ModelProvider` because we need this thing to be a base model of some generated model provider. This applies to TrackResourceData or ResourceData.

but in the SubResource case, we should not use this any more - or at least we should not use it without renaming the above class because SubResource is never inheritable.

I think in our current situation, we should return the model's type in a framework type such as `typeof(SubResource)`.